sdk_extensions/logging/support/src/main/java/io/opentelemetry/logging/SizeOrLatencyBatchStrategy.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.logging;
+
+import io.opentelemetry.logging.api.LogRecord;
+import java.util.Collection;
+import java.util.Queue;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Batching strategy that allows specifying both a maximum size and a maximum amount of time before
+ * sending a batch of records. Batch will be flushed whenever the list of records queued exceeds the
+ * maximum threshold (default 50), or a set timeout expires (default 5 seconds).
+ */
+public class SizeOrLatencyBatchStrategy implements LoggingBatchStrategy {
+  private static final int DEFAULT_BATCH_SIZE = 50;
+  private static final int DEFAULT_MAX_DELAY = 5;
```

Comment:
I am not sure what defaults we use for trace and metrics but I think a lower default would be nicer. Latencies at every node are additive and keeping it low is beneficial otherwise the data may take a lot of time to arrive. We use 200ms in the Collector.